src/geos.c

```diff
@@ -16,6 +16,40 @@ int init_geos(PyObject *m)
     return 0;
 }
 
+
+/* Checks whether the geometry is a multipoint with an empty point in it
+ *
+ * According to https://github.com/libgeos/geos/issues/305, this check is not
+ * necessary for GEOS 3.7.3, 3.8.2, or 3.9. When these versions are out, we 
+ * should add version conditionals and test.
+ * 
+ * The return value is one of:
+ * - PGERR_SUCCESS 
+ * - PGERR_MULTIPOINT_WITH_POINT_EMPTY
+ * - PGERR_GEOS_EXCEPTION
+ */
+char check_to_wkt_compatible(GEOSContextHandle_t ctx, GEOSGeometry *geom) {    
+    int n, i;
+    char geom_type, is_empty;
+    GEOSGeometry *sub_geom;
+
+    geom_type = GEOSGeomTypeId_r(ctx, geom);
+    if (geom_type == -1) { return PGERR_GEOS_EXCEPTION; }
+    if (geom_type != GEOS_MULTIPOINT) { return PGERR_SUCCESS; }
+    
+    n = GEOSGetNumGeometries_r(ctx, geom);
+    if (n == -1) { return PGERR_GEOS_EXCEPTION; }
+    for(i = 0; i < n; i++) {
+        sub_geom = GEOSGetGeometryN_r(ctx, geom, i);
+        if (sub_geom == NULL) { return PGERR_GEOS_EXCEPTION; }
+        is_empty = GEOSisEmpty_r(ctx, sub_geom);
```

Comment:
Not sure if it would be helpful to add as a comment here, but I found the value `2` unexpected until I found this in the GEOS source:
```
Unary predicate - return 2 on exception, 1 on true, 0 on false
```